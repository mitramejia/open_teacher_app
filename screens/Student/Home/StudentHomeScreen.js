// =============================================================
//
// Name: StudentHomeScreen
// -> Description:
//
// Author: mitramejia
// Created at: 8/21/17
//
// =============================================================

import React from 'react';
import { StackNavigator } from 'react-navigation';
import { Content, Text } from 'native-base';
import strings from './Strings';
import styles from './StyleSheet';

class StudentHomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  static navigationOptions = {
    title: 'StudentHomeScreen',
    header: null,
  };

  render() {
    return (
      <Content contentContainerStyle={styles.container}>
        <Text>
          {strings.test}
        </Text>
      </Content>
    );
  }
}

// Export to Navigation
export default StackNavigator({
  StudentHomeScreen: {
    screen: StudentHomeScreen,
  },
});
