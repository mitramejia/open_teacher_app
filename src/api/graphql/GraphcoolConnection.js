// =============================================================
//
// Name: GraphcoolConnection
// -> Description:
//
// Author: mitramejia
// Created at: 8/21/17
//
// =============================================================

import { ApolloClient, createNetworkInterface } from 'apollo-client';
import { GRAPHCOOL_SIMPLE_API, GRAPHCOOL_WEB_SOCKET_CLIENT } from 'react-native-dotenv';
import { addGraphQLSubscriptions, SubscriptionClient } from 'subscriptions-transport-ws';
import { AsyncStorage } from 'react-native';

const networkInterface = createNetworkInterface({ uri: GRAPHCOOL_SIMPLE_API });

// Use middleware to manage auth tokens
networkInterface.use([
  {
    applyMiddleware(req, next) {
      if (!req.options.headers) {
        req.options.headers = {}; // Create the header object if needed.
      }

      AsyncStorage.getItem('token').then(
        encodedToken => {
          req.options.headers['authorization'] = `Bearer ${encodedToken}`;
          next();
        },
        failure => {
          console.error('ERROR: no token', failure);
          next();
        }
      );
    },
  },
]);

// Create WebSocket client
const wsClient = new SubscriptionClient(GRAPHCOOL_WEB_SOCKET_CLIENT, {
  reconnect: true,
});
// Extend the network interface with the WebSocket
const networkInterfaceWithSubscriptions = addGraphQLSubscriptions(networkInterface, wsClient);

const client = new ApolloClient({
  networkInterface: networkInterfaceWithSubscriptions,
  dataIdFromObject: o => o.id,
});

export default client;
