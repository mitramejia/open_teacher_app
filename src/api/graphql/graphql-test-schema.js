// =============================================================
//
// Name: graphql-test-schema
// -> Description:
//
// Author: mitramejia
// Created at: 8/28/17
//
// =============================================================

import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { mockNetworkInterfaceWithSchema } from 'apollo-test-utils';
import { ApolloClient } from 'apollo-client';

export function mockApolloClient() {
  const schema = makeExecutableSchema({ typeDefs });
  addMockFunctionsToSchema({ schema });
  const mockNetworkInterface = mockNetworkInterfaceWithSchema({ schema });
  return new ApolloClient({
    networkInterface: mockNetworkInterface,
  });
}

export const typeDefs = `
      type User {
        id: Int
        firstName: String
        lastName: String
        email: String
      }

      type Query {
        user: User
      }
    `;
