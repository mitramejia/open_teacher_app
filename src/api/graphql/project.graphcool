# project: cj5zswpmnfr6201049umywa1m
# version: 48

type Post implements Node {
  description: String!
  imageUrl: String!
  createdAt: DateTime!
  id: ID! @isUnique
  updatedAt: DateTime!
}

type File implements Node {
  contentType: String!
  createdAt: DateTime!
  id: ID! @isUnique
  name: String!
  secret: String! @isUnique
  size: Int!
  updatedAt: DateTime!
  url: String! @isUnique
}

type Tutor implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  updatedAt: DateTime!
  students: [Student!]! @relation(name: "TutorOnStudent")
  description: String
  diplomaUrl: String!
  lessons: [Lesson!]! @relation(name: "TutorOnLesson")
  locations: [String!]!
  subjects: [Subject!]! @relation(name: "SubjectOnTutor")
  user: User! @relation(name: "UserOnTutor")
}

type User implements Node {
  createdAt: DateTime!
  firstName: String!
  imageUrl: String
  lastName: String!
  id: ID! @isUnique
  updatedAt: DateTime!
  email: String @isUnique
  isTutor: Boolean! @defaultValue(value: false)
  password: String
  student: Student @relation(name: "UserOnStudent")
  tutor: Tutor @relation(name: "UserOnTutor")
}

type Lesson implements Node {
  cost: Float!
  createdAt: DateTime!
  duration: Int!
  id: ID! @isUnique
  student: Student @relation(name: "StudentOnLesson")
  updatedAt: DateTime!
  user: Tutor @relation(name: "TutorOnLesson")
}

type Student implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  lessons: [Lesson!]! @relation(name: "StudentOnLesson")
  subjects: [Subject!]! @relation(name: "SubjectOnStudent")
  tutor: Tutor @relation(name: "TutorOnStudent")
  updatedAt: DateTime!
  user: User! @relation(name: "UserOnStudent")
}

type Subject implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  name: String!
  students: [Student!]! @relation(name: "SubjectOnStudent")
  tutors: [Tutor!]! @relation(name: "SubjectOnTutor")
  updatedAt: DateTime!
}