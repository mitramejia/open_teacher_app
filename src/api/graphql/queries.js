// =============================================================
//
// Name: model
// -> Description:
//
// Author: mitramejia
// Created at: 8/25/17
//
// =============================================================

import { gql } from 'react-apollo';

export const createAccount = gql`
  mutation($firstName: String!, $lastName: String!, $email: String!, $password: String!) {
    createUser(
      authProvider: { email: { email: $email, password: $password } }
      firstName: $firstName
      lastName: $lastName
    ) {
      id
      firstName
      lastName
      email
      password
    }
  }
`;

export const userQuery = gql`
  query {
    user {
      id
      firstName
      lastName
      email
    }
  }
`;

export const signinUser = gql`
  mutation($email: String!, $password: String!) {
    signinUser(email: { email: $email, password: $password }) {
      token
    }
  }
`;

export const allSubjectsQuery = gql`
  query {
    allSubjects {
      id
      name
    }
  }
`;
