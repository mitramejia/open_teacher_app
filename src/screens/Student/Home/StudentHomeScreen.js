// =============================================================
//
// Name: StudentHomeScreen
// -> Description:
//
// Author: mitramejia
// Created at: 8/21/17
//
// =============================================================

import React from 'react';
import {
  Container,
  View,
  Content,
  Header,
  Form,
  Item,
  Input,
  Toast,
  Button,
  Icon,
  StyleProvider,
  connectStyle,
} from 'native-base';
import PropTypes from 'prop-types';
import style from './style';
import strings from './strings';
import { colors } from '../../../config/style';
import OptionPicker from '../../../components/OptionPicker';

class StudentHomeScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    return (
      <Content padder style={style.formContainer}>
        <View style={style.formContainer}>
          <Icon
            style={style.userIcon}
            name="ios-contact-outline"
            size={44}
            color={colors.secondary}
          />
        </View>
        <Form ref="registrationForm">
          <OptionPicker name="Materias" icon="ios-book-outline" data={this.props.data} />
          <Item style={style.inputField} regular>
            <Icon name="ios-calendar-outline" size={28} color={colors.secondary} />
            <Input
              placeholder={strings.timeInputField}
              ref={input => {
                this.time = input;
              }}
            />
          </Item>
        </Form>
      </Content>
    );
  }
}

StudentHomeScreen.propTypes = {
  user: React.PropTypes.object.isRequired,
  data: React.PropTypes.array.isRequired,
};

export default StudentHomeScreen;
