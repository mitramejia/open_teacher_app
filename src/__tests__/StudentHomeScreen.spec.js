import StudentHomeScreen from '../screens/Student/Home/StudentHomeScreen';
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {ApolloProvider} from "react-apollo";
import { mockApolloClient } from '../api/graphql/graphql-test-schema';

it('renders a Login Screen', () => {
  const navigation = {};
  const user = {};

  const tree = renderer
    .create(
      <ApolloProvider client={mockApolloClient()}>
        <StudentHomeScreen navigation={navigation} user={user} />
      </ApolloProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
