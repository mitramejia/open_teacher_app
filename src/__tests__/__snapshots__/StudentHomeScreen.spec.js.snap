// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Login Screen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "backgroundColor": "#eeeeee",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      onScroll={[Function]}
      padder={true}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={0}
      showsVerticalScrollIndicator={true}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "#1D374C",
            "paddingBottom": 30,
            "paddingTop": 24,
          },
        ]
      }
      virtual={undefined}
    >
      <View>
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#1D374C",
                "paddingBottom": 30,
                "paddingTop": 24,
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#5090CD",
                  "fontSize": 44,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": "flex-end",
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
        </View>
        <View
          style={Object {}}
          virtual={undefined}
        >
          <ExponentAppLoading />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FAFCFF",
                "borderBottomWidth": 1,
                "borderColor": "#5090CD",
                "borderLeftWidth": 1,
                "borderRadius": 12,
                "borderRightWidth": 1,
                "borderTopWidth": 1,
                "flexDirection": "row",
                "height": 72,
                "marginBottom": 24,
                "marginLeft": 0,
                "opacity": 1,
                "paddingLeft": 24,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#5090CD",
                    "fontSize": 28,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                      "paddingLeft": 10,
                      "paddingRight": 8,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              
            </Text>
            <TextInput
              editable={true}
              placeholder="Cuándo estás disponible?"
              placeholderTextColor="#575757"
              style={
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "lineHeight": 24,
                  "paddingLeft": 8,
                  "paddingRight": 5,
                  "top": 1.5,
                }
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              virtual={undefined}
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
